app/com/linkedin/drelephant/analysis/AnalyticJobGeneratorHadoop2.java

```diff
@@ -198,21 +219,31 @@ private JsonNode readJsonNode(URL url)
    * @throws IOException
    * @throws AuthenticationException Problem authenticating to resource manager
    */
-  private List<AnalyticJob> readApps(URL url) throws IOException, AuthenticationException{
+  private JsonNode readApps(URL url) throws IOException, AuthenticationException {
     List<AnalyticJob> appList = new ArrayList<AnalyticJob>();
-
     JsonNode rootNode = readJsonNode(url);
-    JsonNode apps = rootNode.path("apps").path("app");
+    return rootNode.path("apps").path("app");
+  }
 
+  /**
+   * When called first time after launch, hit the DB and avoid duplicated analytic jobs that have been analyzed
+   * before.
+   * @param apps The list of apps in JsonNode
+   * @param lastTime The upper limit of the last fetch interval
+   * @return The filtered list of analytic jobs
+   */
+  private List<AnalyticJob> filterDuplicates(JsonNode apps, long lastTime) {
+    List<AnalyticJob> appList = new ArrayList<AnalyticJob>();
     for (JsonNode app : apps) {
       String appId = app.get("id").getValueAsText();
-
-      // When called first time after launch, hit the DB and avoid duplicated analytic jobs that have been analyzed
-      // before.
-      if (_lastTime > 0 || (_lastTime == 0 && AppResult.find.byId(appId) == null)) {
+      if (lastTime > 0 || (lastTime == 0 && AppResult.find.byId(appId) == null) ||
```

Comment:
Can you add comment explaining what this condition is looking for readability purpose ? Or better yet, create micro functions for each condition and use them instead.